internal/provisioner/cluster_provisioning.go

```diff
@@ -26,11 +28,38 @@ import (
 func provisionCluster(
 	cluster *model.Cluster,
 	kubeconfigPath string,
+	tempDir string,
 	awsClient aws.AWS,
+	gitClient git.Client,
+	argocdClient argocd.Client,
 	params ProvisioningParams,
 	store model.ClusterUtilityDatabaseStoreInterface,
 	logger logrus.FieldLogger) error {
 
+	// Register cluster in argocd
+	switch {
+	case cluster.UtilityMetadata.ManagedByArgocd:
+		switch {
+		case !cluster.UtilityMetadata.ArgocdClusterRegister.Registered:
+			logger.Debug("Starting argocd cluster registration")
+			clusterRegister, err := NewClusterRegisterHandle(cluster, gitClient, awsClient.GetCloudEnvironmentName(), tempDir, logger)
+			if err != nil {
+				return errors.Wrap(err, "Failed to create new cluster register handle")
+			}
+			if err = clusterRegister.clusterRegister(params.S3StateStore); err != nil {
+				return errors.Wrap(err, "failed to register cluster in argocd")
+			}
+
+			cluster.UtilityMetadata.ArgocdClusterRegister.Registered = true
+			logger.Infof("Cluster: %s successfully registered in argocd", cluster.ID)
+
+		default:
+			logger.Info("Cluster already registered, skipping argocd cluster register")
+		}
+	default:
+		logger.Info("ManagedByArgocd is false, skipping argocd cluster registration")
+	}
```

Comment:
Any particular reason to use a switch here, rather than just an if/else? You could probably narrow down to one level as well - adding the `ManagedByArgocd` value to the logger fields will allow you to communicate that information via logs, then you can just check `cluster.UtilityMetdata.ManagedByArgocd && !cluster.UtilityMetadata.ArgocdClusterRegister.Registered`

Reply:
No particular reason I was trying to keep it simple but your suggestion is better. I changed it. Thank you @nickmisasi